torch/utils/_pytree.py

```diff
@@ -470,14 +470,23 @@ def _is_leaf(tree: PyTree) -> bool:
 # context: some context that is useful in unflattening the pytree
 # children_specs: specs for each child of the root Node
 # num_leaves: the number of leaves
-@dataclasses.dataclass
+@dataclasses.dataclass(init=True, frozen=True, eq=True, repr=False)
```

Comment:
Dynamo does not support frozen dataclasses.

It errors for statement:

```python
return TreeSpec(node_type, context, children_specs)
```

The error:

```text
torch._dynamo.exc.Unsupported: call_function UserDefinedClassVariable() [BuiltinVariable(list), ConstantVariable(NoneType), ListVariable()] {}
```

A minimal script to reproduce:

```python
import dataclasses

import torch


@dataclasses.dataclass(frozen=True)
class A:
    a: int


def fn() -> None:
    a = A(1)


comp_out = torch._dynamo.optimize(nopython=True)(fn)()
```

The error for statement `a = A(1)`:

```text
torch._dynamo.exc.Unsupported: call_function UserDefinedClassVariable() [ConstantVariable(int)] {}
```